packages/stark-ui/src/modules/transform-input-directive/directives/transform-input.directive.spec.ts

```diff
@@ -0,0 +1,385 @@
+
+			expect(component.value).toBe(expected);
+		});
+	});
+
+	describe("with formControl", () => {
+		@Component({
+			selector: "test-component",
+			template: "<input [formControl]='formControl' [starkTransformInput]='starkTransformInputValue'/>"
+		})
+		class TestComponent {
+			public starkTransformInputValue: StarkInputTransformationType = () => {
+				/*noop*/
+			};
+			public formControl: FormControl = new FormControl("");
+
+			public constructor() {
+				this.formControl.valueChanges.subscribe(this.changeObserver);
+			}
+
+			public changeObserver(): void {
+				/*Placeholder for spy*/
+			}
+		}
+
+		let fixture: ComponentFixture<TestComponent>;
+		let component: TestComponent;
+		let mockValueChangeObserver: jasmine.SpyObj<Observer<any>>;
+		let htmlInputElement: HTMLInputElement;
+
+		beforeEach(() => {
+			TestBed.configureTestingModule({
+				declarations: [StarkTransformInputDirective, TestComponent],
+				imports: [ReactiveFormsModule]
+			});
+
+			fixture = TestBed.createComponent(TestComponent);
+			component = fixture.componentInstance;
+			htmlInputElement = fixture.nativeElement.querySelector("input");
+
+			// Register mock subscription
+			mockValueChangeObserver = jasmine.createSpyObj<Observer<any>>("observerSpy", ["next", "error", "complete"]);
+			component.formControl.valueChanges.subscribe(mockValueChangeObserver);
+
+			// trigger initial data binding
+			fixture.detectChanges();
+
+			expect(component.formControl.value).toBe("", "field 'value' should start as empty string ");
+		});
+
+		it("should set value to uppercase", () => {
+			const input: string = "upper";
+			const expected: string = "UPPER";
+
+			// Set directive to correct implementation
+			component.starkTransformInputValue = "uppercase";
+			fixture.detectChanges();
+
+			mockInputEvent(input, htmlInputElement);
+			fixture.detectChanges();
+
+			expect(component.formControl.value).toBe(expected);
+			expect(mockValueChangeObserver.next).toHaveBeenCalledTimes(1);
+			expect(mockValueChangeObserver.error).not.toHaveBeenCalled();
+			expect(mockValueChangeObserver.complete).not.toHaveBeenCalledTimes(1);
+		});
+
+		it("should set value to lowercase", () => {
+			const input: string = "LOWER";
+			const expected: string = "lower";
+
+			// Set directive to correct implementation
+			component.starkTransformInputValue = "lowercase";
+			fixture.detectChanges();
+
+			mockInputEvent(input, htmlInputElement);
+			fixture.detectChanges();
+
+			expect(component.formControl.value).toBe(expected);
+			expect(mockValueChangeObserver.next).toHaveBeenCalledTimes(1);
+			expect(mockValueChangeObserver.error).not.toHaveBeenCalled();
+			expect(mockValueChangeObserver.complete).not.toHaveBeenCalledTimes(1);
+		});
+
+		it("should replace dirty word", () => {
+			const input: string = "fudge you!";
+			const expected: string = "***** you!";
+
+			// Set directive to correct implementation
+			component.starkTransformInputValue = (v: string) =>
```

Comment:
You could avoid duplicating this function if you declare it at the top of this file and use it in the different tests where you need it